### PR TITLE
Issue #306: Fix and Enforce Formatting

### LINKED ISSUE #306: Add ESLint and Prettier to CI/CD Pipeline
Add workflows to surface visibility of ESLint and Prettier issues on pull request creation

### PR DESCRIPTION
**Pull request by kai-thompson**
closes #306
NOTE: This is an optional PR. I do not care if this is rejected.

Changes:
- Addresses all ESLint warnings and errors
- Fix formatting
- Adds Prettier and ESLint checks to CI/CD pipeline

### CODE REVIEW

**Review by jahabeebs — CHANGES_REQUESTED** [review]
Thanks for cleaning the repo up 

- Let's change node-version in the ci-pr to 20 since we use this for the app itself. I don't mind having lint/formatting stuff in the CI as long as it doesn't block the PR from being merged (though maybe Pat would have a different opinion)
- An infinite loop was introduced somewhere in this PR that is preventing me from accessing any vaults in the testnet deployment (click into the Vercel testnet deployment in the PR and you'll see). Make sure you're able to enter into your vaults and deposit/withdraw without issues. If you need testnet tokens you can use the /claim command with your wallet address in the bot channel in our public Discord but we should test the basic functionality

Infinite loop

Screenshot 2024-03-19 at 1 38 43 PM

Can't see vaults (sometimes it'll flicker the vault into view but very sporadically)

https://github.com/open-dollar/od-app/assets/47253537/66201a04-1ebf-4a61-8fc9-e4854e659059

**Comment by kai-thompson** [comment]
> Thanks for cleaning the repo up
> 
> * Let's change node-version in the ci-pr to 20 since we use this for the app itself. I don't mind having lint/formatting stuff in the CI as long as it doesn't block the PR from being merged (though maybe Pat would have a different opinion)
> * An infinite loop was introduced somewhere in this PR that is preventing me from accessing any vaults in the testnet deployment (click into the Vercel testnet deployment in the PR and you'll see). Make sure you're able to enter into your vaults and deposit/withdraw without issues. If you need testnet tokens you can use the /claim command with your wallet address in the bot channel in our public Discord but we should test the basic functionality
> 
> Infinite loop
> 
> Screenshot 2024-03-19 at 1 38 43 PM
> Can't see vaults (sometimes it'll flicker the vault into view but very sporadically)
> 
>  Screen.Recording.2024-03-19.at.1.37.57.PM.mov

Just removed all but one (not risky) useEffect dependency array changes!

**Review by jahabeebs — APPROVED** [review]

File: src/services/TransactionUpdater.tsx [hunk 1]
```diff
@@ -1,6 +1,4 @@
-import React, { useEffect, useMemo } from 'react'
-import { toast } from 'react-toastify'
```
Comment (pi0neerpat) [inline src/services/TransactionUpdater.tsx:2]: Lets leave this library in for now, since we might use it for bridged asset notifications

**Review by pi0neerpat — CHANGES_REQUESTED** [review]
Getting error
```js
Cannot read properties of undefined (reading 'charAt')
TypeError: Cannot read properties of undefined (reading 'charAt')
    at capitalizeName (http://localhost:3000/static/js/bundle.js:34642:37)
    at networkChecker (http://localhost:3000/static/js/bundle.js:24194:95)
    at http://localhost:3000/static/js/bundle.js:24223:5
    ```

**Comment by kai-thompson** [comment]
> import { toast } from 'react-toastify'



> Getting error
> 
> ```js
> Cannot read properties of undefined (reading 'charAt')
> TypeError: Cannot read properties of undefined (reading 'charAt')
>     at capitalizeName (http://localhost:3000/static/js/bundle.js:34642:37)
>     at networkChecker (http://localhost:3000/static/js/bundle.js:24194:95)
>     at http://localhost:3000/static/js/bundle.js:24223:5
>     ```
> ```

@pi0neerpat  Strange. What environment is this from? It seems to be working as expected locally for me. What're some steps I can take to replicate this?

**Review by pi0neerpat — APPROVED** [review]
The error was due to testing on Optimism, which only works on branch `optimism-hai`

**Pull request merged**